sql/src/main/java/io/crate/operation/projectors/BatchConsumerToRowReceiver.java

```diff
@@ -103,4 +133,13 @@ private RepeatHandle getRepeatHandle(BatchIterator iterator) {
             consumeIterator(iterator);
         };
     }
+
+    @Override
+    public void kill(@Nullable Throwable throwable) {
+        if (running) {
+            rowReceiver.fail(throwable);
+        } else {
+            accept(null, throwable);
```

Comment:
imho there is a race condition, when kill is called while accept is processing and running is still false, however it is the same as in the other case. the kill is not required to be propagated to the consumer, since the consumer will be killed anyways